File: apps/mybookkeeper/frontend/src/shared/types/tax/tax-year-profile.ts
export interface TaxYearProfile {
  id: string;
  organization_id: string;
  tax_year: number;
  filing_status: string | null;
  dependents_count: number;
  property_use_days: Record<string, { personal_days: number; rental_days: number }>;
  home_office_sqft: number | null;
  home_total_sqft: number | null;
  business_mileage: number | null;
}

File: apps/mybookkeeper/backend/app/schemas/tax/tax_year_profile.py
import uuid
from typing import Literal

from pydantic import BaseModel, Field


FilingStatus = Literal["single", "married_filing_jointly", "married_filing_separately", "head_of_household", "qualifying_surviving_spouse"]


class TaxYearProfileUpdate(BaseModel):
    filing_status: FilingStatus | None = None
    dependents_count: int | None = Field(default=None, ge=0, le=20)
    property_use_days: dict[str, int] | None = None
    home_office_sqft: int | None = None
    home_total_sqft: int | None = None
    business_mileage: int | None = None


class TaxYearProfileRead(BaseModel):
    id: uuid.UUID
    organization_id: uuid.UUID
    tax_year: int
    filing_status: str | None
    dependents_count: int
    property_use_days: dict[str, int]
    home_office_sqft: int | None
    home_total_sqft: int | None
    business_mileage: int | None

    model_config = {"from_attributes": True}
